geniza/corpus/models.py

```diff
@@ -580,7 +586,6 @@ def items_to_index(cls):
             "tags",
             "languages",
             "footnotes",
-            "log_entries",
```

Comment:
I don't think this is only place we're prefetching log entries? We can live with the additional time for bulk indexing, but I'm concerned about the impact for the admin csv download.

Reply:
You're right, that's the other place where it's used. I imagine it's for this:

https://github.com/Princeton-CDH/geniza/blob/39e5be7335b8479f762455a22ce3b12e8cdc13b1/geniza/corpus/admin.py#L325-L329

I can see how this might result in a lot of unnecessarily hitting the DB without prefetching `log_entries`. I thought about just doing a simple queryset before the loop to get all matching log entries for all these documents, but that still would result in additional queries on filtering. I read through [this](https://www.mattduck.com/2021-01-django-orm-result-cache.html) but nothing there seemed particularly helpful. Should we consider this a blocker?

Reply:
It's a blocker if it makes the admin download unusable (which I suspect it does); we'll need to test the performance. There is probably a way to customize the prefetching, or maybe we can use a filtered relation https://docs.djangoproject.com/en/3.2/ref/models/querysets/#filteredrelation-objects

Reply:
I'm stuck on both of these options. I can't seem to get a way to use any of `prefetch_related`, `select_related`, `prefetch_related_objects`, `Prefetch`, or `FilteredRelation` without needing to pass a field name for the `Document` model that relates it directly to `LogEntry`.

The best I can come up with so far is doing the following before running the loop over documents in the CSV export:
```py
doc_ids = queryset.values_list("id", flat=True)
prefetched_log_entries = LogEntry.objects.filter(
    object_id__in=doc_ids,
    content_type__app_label="corpus",
    content_type__model="document",
)
```

But I'm still blocked there because any additional filtering on `prefetched_log_entries` will still result in the same additional DB queries as without this.